content/contributing/documentation/rst_guidelines.rst

```diff
@@ -1,162 +1,1324 @@
+:custom-css: showcase_tables.css
+
 ==============
 RST guidelines
 ==============
 
-.. _contributing/relative-links:
+.. _contributing-rst-formatting:
+
+Use formatting
+==============
+
+Use specific formatting to improve clarity and readability. For example, apply
+:ref:`contributing-rst-menuselection` for menu paths, :ref:`contributing-rst-guilabel`
+for other user interface elements, such as fields, buttons and options, :ref:`contributing-rst-note`
+for notes, :ref:`contributing-rst-example` for examples, etc.
+
+.. note::
+   Add a blank line between different block elements, such as paragraphs, lists, and directives to
+   ensure proper rendering and formatting.
+
+Hyperlinks
+==========
+
+.. _contributing-rst-relative-links:
 
 Use relative links for internal URLs
-====================================
+------------------------------------
 
-If you need to reference an internal documentation page or a file that is not sitting in the same
-directory as your current page, always make use of *relative file paths* rather than *absolute file
-paths*. An absolute file path indicates the location of the target from the root of its file tree. A
-relative file path makes use of smart notations (such as ``../`` that redirects to the parent
-folder) to indicate the location of the target *relative* to that of the source document.
+If you need to reference an :ref:`internal documentation page <contributing-rst-doc-page-hyperlinks>`
+or a :ref:`file <contributing-rst-file>` that is not located in the same directory as
+your current page, always use *relative file paths* instead of *absolute file paths*. This ensures
+that links remain valid even with version updates, folder name changes, and directory structure
+reorganization.
 
-Example
--------
+An absolute file path indicates target's location from the root directory. A relative file path uses
+smart notations (such as ``../`` that redirects to the parent folder) to indicate the target's
+location *relative* to that of the source document.
+
+.. example::
+   Given the following source file tree:
+   ::
+
+     documentation
+     ├── content
+     │   └── applications
+     │   │   └── sales
+     │   │   │   └── sales
+     │   │   │   │   └── products_prices
+     │   │   │   │   │   └── products
+     │   │   │   │   │   │   └── import.rst
+     │   │   │   │   │   │   └── variants.rst
+     │   │   │   │   │   └── prices.rst
+
+   A reference to the rendered :file:`prices.html` and :file:`variants.html` could be made from
+   :file:`import.rst` as follows:
+
+   #. Absolute:
+
+      - ``https://odoo.com/documentation/{BRANCH}/applications/sales/sales/products_prices/prices.html``
+      - ``https://odoo.com/documentation/{BRANCH}/applications/sales/sales/products_prices/products/variants.html``
+
+   #. Relative:
+
+      - ``../prices.html``
+      - ``variants.html``
+
```

Comment:
I thought about using an RST file as an example, but you should be using the `:doc:` markup then if you follow the guidelines :) Also, relative links work for all file types.

Reply:
Yeah, I thought it would be fine just for this example but you're right, we don't want to add confusion. I still think this is the most helpful example (since that's the one we use relative links for the most, and people struggle a lot with this) so I just added the :doc: markup, what do you think?